### PR TITLE
pbc-sig: fix flat namespace usage

### PR DESCRIPTION
**Pull request by danielnachun**
- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [ ] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by danielnachun** [comment]
I'm puzzled as to why this is now failing on ARM when it was previously bottled on ARM Big Sur.  I looked at the logs for the failed build on ARM Big Sur and these are the relevant lines from `config.log`:
```
configure:11370: clang -o conftest -fnested-functions  -L/Library/Developer/CommandLineTools/SDKs/MacOSX11.sdk/usr/lib conftest.c -lm   >&5
conftest.c:30:6: warning: incompatible redeclaration of library function 'pow' [-Wincompatible-library-redeclaration]
char pow ();
     ^
conftest.c:30:6: note: 'pow' is a builtin with type 'double (double, double)'
1 warning generated.
ld: the target architecture doesn't support executable stacks
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```
One of the only mentions I can find of this is here: https://github.com/libffi/libffi/issues/695, but in that case the error only occurs because `-Wl,-allow_stack_execute` was being explicitly passed as an argument to the linker, which we are **not** doing in this case.

**Comment by danielnachun** [comment]
@Homebrew/core this fails to build on ARM even when I just push a comment with no changes.  I don't know how this worked before on ARM but clearly it doesn't now.  Should we debottle this on ARM Big Sur?

**Comment by cho-m** [comment]
I'm guessing it is failing due to `-fnested-functions` added in `configure(.ac)`. Could be some change in build environment between when Big Sur ARM was previously bottled and now.

**Comment by danielnachun** [comment]
Weirdly, it looks like this is just generically added to any `darwin` host:
```
case $host_os in
     darwin*)
              dnl nested function support
              AC_MSG_NOTICE([Adding -fnested-functions to CFLAGS])
              CFLAGS="-fnested-functions"
```
Nonetheless I'll try blanking this on ARM to see if that fixes it.

File: Formula/pbc-sig.rb [hunk 1]
```diff
@@ -21,10 +21,19 @@ class PbcSig < Formula
...
+    sha256 "83af02f2aa2b746bb7225872cab29a253264be49db0ecebb12f841562d9a2923"
+  end
+
   # https://groups.google.com/forum/#!topic/pbc-devel/ZmFCHZmrhcw
   patch :DATA
 
   def install
+    # Disable -fnested-functions CFLAG on ARM, which will cause it to fail with:
+    # incompatible redeclaration of library function 'pow'
+    inreplace "configure.ac", "-fnested-functions", "" if Hardware::CPU.arm?
```
Comment (cho-m) [inline Formula/pbc-sig.rb:36]: This will need to be done in `configure` to do anything as build isn't regenerating it.

Comment (danielnachun) [reply Formula/pbc-sig.rb:36]: Whoops, I'll fix that.

**Comment by danielnachun** [comment]
Thanks for fix @cho-m, it's working now!

File: Formula/pbc-sig.rb [hunk 2]
```diff
@@ -21,10 +21,19 @@ class PbcSig < Formula
...
+    sha256 "83af02f2aa2b746bb7225872cab29a253264be49db0ecebb12f841562d9a2923"
+  end
+
   # https://groups.google.com/forum/#!topic/pbc-devel/ZmFCHZmrhcw
   patch :DATA
 
   def install
+    # Disable -fnested-functions CFLAG on ARM, which will cause it to fail with:
+    # incompatible redeclaration of library function 'pow'
+    inreplace "configure", "-fnested-functions", "" if Hardware::CPU.arm?
```
Comment (SMillerDev) [inline Formula/pbc-sig.rb:36]: This should be reported upstream

**Comment by MikeMcQuaid** [comment]
> this fails to build on ARM even when I just push a comment with no changes. I don't know how this worked before on ARM but clearly it doesn't now. Should we debottle this on ARM Big Sur?

@danielnachun I have no opinions on this formula specifically but in this case: definitely not. If something doesn't build but there is a working bottle: great, then it works for most of our users.

**Comment by danielnachun** [comment]
> > this fails to build on ARM even when I just push a comment with no changes. I don't know how this worked before on ARM but clearly it doesn't now. Should we debottle this on ARM Big Sur?
> 
> @danielnachun I have no opinions on this formula specifically but in this case: definitely not. If something doesn't build but there is a working bottle: great, then it works for most of our users.

Good to know.  I was worried the bottle was actually broken for users but fortunately in this case it was not!

File: Formula/pbc-sig.rb (hunk 2 above)
Comment (danielnachun) [reply Formula/pbc-sig.rb]: While there is no bug tracker, there is a mailing list with a public archive so I've sent an email there and posted the link in a comment.

**Review by SMillerDev — APPROVED** [review]

**Pull request closed without merging**